examples/notebooks/Splitters Tutorial.ipynb

```diff
@@ -0,0 +1,370 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import deepchem as dc\n",
+    "import pandas as pd\n",
+    "import os"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We start with the IndexSplitter. This splitter returns a range object which contains the split according to the fractions provided by the user. The three range objects can then be used to iterate over the dataset as test,valid and Train.\n",
+    "\n",
+    "\n",
+    "#Each of the splitters that will be used has two functions inherited from the main class that are `train_test_split` which can be used to split the data into training and tesing data and the other fucnction is `train_valid_test_split` which is used to split the data to train, validation and test split.\n",
+    "\n",
+    "#Note: All the splitters have a default percentage of 80,10,10 as train, valid and test respectively. But can be changed by specifying the `frac_train`,`frac_test` and `frac_valid` in the ratio we want to split the data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 31,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "input_data=os.path.join(\"/home/ab/deepchem/deepchem/models/tests/example.csv\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We then featurize the data using any one of the featurizers present."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Loading raw samples now.\n",
+      "shard_size: 8192\n",
+      "About to start loading CSV from /home/ab/deepchem/deepchem/models/tests/example.csv\n",
+      "Loading shard 1 of size 8192.\n",
+      "Featurizing sample 0\n",
+      "TIMING: featurizing shard 0 took 0.031 s\n",
+      "TIMING: dataset construction took 0.040 s\n",
+      "Loading dataset from disk.\n"
+     ]
+    }
+   ],
+   "source": [
+    "tasks=['log-solubility']\n",
+    "featurizer=dc.feat.CircularFingerprint(size=1024)\n",
+    "loader = dc.data.CSVLoader(tasks=tasks, smiles_field=\"smiles\",featurizer=featurizer)\n",
+    "dataset=loader.featurize(input_data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from deepchem.splits.splitters import IndexSplitter"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "splitter=IndexSplitter()\n",
+    "train_data,valid_data,test_data=splitter.split(dataset)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(range(0, 8), range(8, 9), range(9, 10))"
```

Comment:
I think it would be clearer if you showed the length before this.  Also if you wrapped the datasets in a len()